crypto/evp/evp_enc.c

```diff
@@ -152,9 +152,8 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
  skip_to_init:
 #endif
     /* we assume block size is a power of 2 in *cryptUpdate */
-    OPENSSL_assert(ctx->cipher->block_size == 1
-                   || ctx->cipher->block_size == 8
-                   || ctx->cipher->block_size == 16);
+    OPENSSL_assert(ctx->cipher->block_size);
+    OPENSSL_assert(!(ctx->cipher->block_size & (ctx->cipher->block_size - 1)));
```

Comment:
We tend not to like `OPENSSL_assert` these days because it causes the library to crash even in production builds. I prefer `ossl_assert` (crash in debug builds, return an error in production builds). Or failing that just `assert`.

Style nit: Please use "== 0" rather than "!" here.